transaction-ballerina/tests/transaction_with_setrollbackonly_test.bal

```diff
@@ -51,3 +51,95 @@ transactional function setRollbackOnlyErrorInTrx() {
     error cause = error("setRollbackOnly");
     trx:setRollbackOnly(cause);
 }
+
+@test:Config {
+}
+function testSetRollbackOnlyWithinLoop() {
+    error? setRollbackOnlyRes = setRollbackOnlyWithinLoop();
+    if (setRollbackOnlyRes is error) {
+        test:assertEquals("setRollbackOnly", setRollbackOnlyRes.message());
+    } else {
+        panic error("Expected an error");
+    }
+}
+
+function setRollbackOnlyWithinLoop() returns error? {
+    string str = "";
+    foreach var i in 1...5 {
+        transaction {
+            str += i.toString();
+            setRollbackOnlyErrorInTrx();
+            check commit;
+            str += i.toString();
```

Comment:
What is the purpose of this variable, I think we have to do some assertion here to test